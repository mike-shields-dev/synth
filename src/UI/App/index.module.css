.App {
    display: grid;
    grid-template-columns: 1fr;
    grid-template-rows: 2rem minmax(0, 1fr) 4rem;
    grid-template-areas: 
        'header'
        'main'
        'footer'
    ;
    height: calc(100vh - 2rem);
    gap: 0.5rem;
}

.App__Header {
    grid-area: header;
}

.App__Main {
    grid-area: main;
    overflow-y: auto;
    overflow-x: hidden;
    scroll-behavior: smooth;
}

.App__Footer {
    grid-area: footer;
}
