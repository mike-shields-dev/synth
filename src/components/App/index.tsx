import React from 'react';
import { useSessionStorage } from '../../hooks/useSessionStorage';
import { Synth } from '../Synth';
import { SynthParameterGroup } from '../SynthParameterGroup';
import './index.module.css';

function App() {
  const [focus, setFocus] = useSessionStorage('focus', "");

  function updateFocus(id: string) {
    setFocus(id);
  } 

  return (
    <div className="App">
      <Synth>
        <SynthParameterGroup 
          groupName="filter" 
          updateFocus={updateFocus}
          isFocused={focus === "filter"}
        >
          <input type="range" />
        </SynthParameterGroup>

        <SynthParameterGroup 
          groupName="filterEnvelope"
          updateFocus={updateFocus}
          isFocused={focus === "filterEnvelope"}
        >
          <input type="range" />
        </SynthParameterGroup>
      </Synth> 
    </div>
  )
}

export default App
