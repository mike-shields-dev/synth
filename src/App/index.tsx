import { useState } from 'react';
import { AmpEnvelope } from '../components/AmpEnv';
import { Filter } from '../components/Filter';
import { FilterEnvelope } from '../components/FilterEnvelope';
import { Oscillator } from '../components/Oscillator';
import { SynthUI } from '../components/SynthUI';
import { useMidiAccess } from '../hooks/useMidiAccess';
import './index.module.css';

function App() {
  const [focus, setFocus] = useState('oscillator');
  useMidiAccess();

  function updateFocus(id: string) {
    setFocus(id);
  }

  return (
    <>
    <div className="App">
      <SynthUI>
        <Oscillator {...{ focus, updateFocus }} />
        <Filter {...{ focus, updateFocus }} />
        <FilterEnvelope {...{ focus, updateFocus }} />
        <AmpEnvelope {...{ focus, updateFocus }} />
      </SynthUI>
    </div>
    </>
  )
}

export default App


