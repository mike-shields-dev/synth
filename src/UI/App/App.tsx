import { useState } from 'react';
import Nav from '../components/Nav';
import { AmpEnvelope, FilterEnvelope, Oscillator, SynthUI } from '../components';
import { Filter } from '../components/Filter';
import { useMidiAccess } from '../hooks/useMidiAccess';
import './index.module.css';
import { publishFocusChange } from '../../PubSub/FocusChange';
import { UIKeyboard } from '../components/UIKeyboard';
import css from './App.module.css';

function App() {
  const [focus, setFocus] = useState('oscillator');
  useMidiAccess();

  function updateFocus(focus: string) {
    setFocus(focus);
    publishFocusChange(focus);
  }

  return (
    <>
      <div className={css.App}>

        <div className={css.App__Header}>
          <Nav {...{ focus, updateFocus }} />
        </div>

        <div className={css.App__Main}>
          <SynthUI>
            <Oscillator {...{ focus, updateFocus }} />
            <Filter {...{ focus, updateFocus }} />
            <FilterEnvelope {...{ focus, updateFocus }} />
            <AmpEnvelope {...{ focus, updateFocus }} />
          </SynthUI>
        </div>

        <div className={css.App__Footer}>
          <UIKeyboard />
        </div>
      </div>
    </>
  )
}

export default App
