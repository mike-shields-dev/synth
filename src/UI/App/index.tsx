import { useState } from 'react';
import { AmpEnvelope, FilterEnvelope, Oscillator, SynthUI } from '../components';
import { Filter } from '../components/Filter';
import { useMidiAccess } from '../hooks/useMidiAccess';
import './index.module.css';
import { publishFocusChange } from '../../PubSub/FocusChange';
import { UIKeyboard } from '../components/UIKeyboard';


function App() {
  const [focus, setFocus] = useState('oscillator');
  useMidiAccess();

  function updateFocus(focus: string) {
    setFocus(focus);
    publishFocusChange(focus);
  }

  return (
    <>
      <div className="App">
        <SynthUI>
          <Oscillator {...{ focus, updateFocus }} />
          <Filter {...{ focus, updateFocus }} />
          <FilterEnvelope {...{ focus, updateFocus }} />
          <AmpEnvelope {...{ focus, updateFocus }} />
          <UIKeyboard />
        </SynthUI>
      </div>
    </>
  )
}

export default App
