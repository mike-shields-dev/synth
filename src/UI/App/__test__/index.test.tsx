import { render, screen } from '@testing-library/react';
import App from '../index';

describe('App', () => {
    beforeEach(() => {
        render(<App />)
    });

    it('matches snapshot', () => {
        const { container } = render(<App />);

        expect(container.firstChild).toMatchSnapshot();
    });

    it('renders the SynthUI', () => {
        expect(screen.getByRole('form', {name: 'synth controls'})).toBeInTheDocument();
    });

    it('renders a SynthParameterGroup for the oscillator', () => {
        expect(screen.getByRole('region', { name: 'Oscillator' })).toBeInTheDocument();
    });

    it('renders a SynthParameterGroup for the filter', () => {
        expect(screen.getByRole('region', { name: 'Filter' })).toBeInTheDocument();
    });

    it('renders a SynthParameterGroup for the filter', () => {
        expect(screen.getByRole('region', { name: 'Filter Envelope' })).toBeInTheDocument();
    });

    it('renders a SynthParameterGroup form the (amp) envelope', () => {
        expect(screen.getByRole('region', { name: 'Amp Envelope' })).toBeInTheDocument();
    })
});