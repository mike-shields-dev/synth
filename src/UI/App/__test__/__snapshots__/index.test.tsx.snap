// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`App > matches snapshot 1`] = `
<div
  class="App"
>
  <form
    aria-label="synth user interface"
  >
    <section
      aria-label="Oscillator"
      class="_SynthParameterGroup--focus_a92880"
      id="oscillator"
    >
      <header>
        <h2>
          Oscillator
        </h2>
      </header>
    </section>
    <section
      aria-label="Filter"
      class="_SynthParameterGroup_a92880"
      id="filter"
    >
      <header>
        <h2>
          Filter
        </h2>
      </header>
      <div>
        <label
          for="filter-frequency"
        >
          Frequency
        </label>
        <input
          id="filter-frequency"
          max="127"
          min="0"
          step="0.05"
          type="range"
          value="-22.58979807604313"
        />
        <output
          aria-label="Filter frequency"
          for="filter-frequency"
        >
          -3.552713678800501e-15
        </output>
      </div>
      <div>
        <label
          for="filter-resonance"
        >
          Resonance
        </label>
        <input
          id="filter-resonance"
          max="127"
          min="0"
          step="0.05"
          type="range"
          value="0"
        />
        <output
          aria-label="Filter resonance"
          for="filter-resonance"
        >
          0
        </output>
      </div>
    </section>
    <section
      aria-label="Filter Envelope"
      class="_SynthParameterGroup_a92880"
      id="filterEnvelope"
    >
      <header>
        <h2>
          Filter Envelope
        </h2>
      </header>
      <div>
        <label
          for="filterEnvelope-attack"
        >
          Attack
        </label>
        <input
          id="filterEnvelope-attack"
          max="127"
          min="0"
          step="0.05"
          type="range"
          value="0"
        />
        <output
          aria-label="Filter Envelope attack"
          for="filterEnvelope-attack"
        >
          0
        </output>
      </div>
      <div>
        <label
          for="filterEnvelope-decay"
        >
          Decay
        </label>
        <input
          id="filterEnvelope-decay"
          max="127"
          min="0"
          step="0.05"
          type="range"
          value="0"
        />
        <output
          aria-label="Filter Envelope decay"
          for="filterEnvelope-decay"
        >
          0
        </output>
      </div>
      <div>
        <label
          for="filterEnvelope-sustain"
        >
          Sustain
        </label>
        <input
          id="filterEnvelope-sustain"
          max="127"
          min="0"
          step="0.05"
          type="range"
          value="0"
        />
        <output
          aria-label="Filter Envelope sustain"
          for="filterEnvelope-sustain"
        >
          0
        </output>
      </div>
      <div>
        <label
          for="filterEnvelope-release"
        >
          Release
        </label>
        <input
          id="filterEnvelope-release"
          max="127"
          min="0"
          step="0.05"
          type="range"
          value="-0.0025400508010160203"
        />
        <output
          aria-label="Filter Envelope release"
          for="filterEnvelope-release"
        >
          -1.3552527156068805e-20
        </output>
      </div>
      <div>
        <label
          for="filterEnvelope-amount"
        >
          Amount
        </label>
        <input
          id="filterEnvelope-amount"
          max="127"
          min="0"
          step="0.05"
          type="range"
          value="0"
        />
        <output
          aria-label="Filter Envelope amount"
          for="filterEnvelope-amount"
        >
          0
        </output>
      </div>
    </section>
    <section
      aria-label="Amp Envelope"
      class="_SynthParameterGroup_a92880"
      id="ampEnvelope"
    >
      <header>
        <h2>
          Amp Envelope
        </h2>
      </header>
      <div>
        <label
          for="envelope-attack"
        >
          Attack
        </label>
        <input
          id="envelope-attack"
          max="127"
          min="0"
          step="0.05"
          type="range"
          value="0.8980256121069153"
        />
        <output
          aria-label="Envelope attack"
          for="envelope-attack"
        >
          0.0001
        </output>
      </div>
      <div>
        <label
          for="envelope-decay"
        >
          Decay
        </label>
        <input
          id="envelope-decay"
          max="127"
          min="0"
          step="0.05"
          type="range"
          value="31.75"
        />
        <output
          aria-label="Envelope decay"
          for="envelope-decay"
        >
          0.5
        </output>
      </div>
      <div>
        <label
          for="envelope-sustain"
        >
          Sustain
        </label>
        <input
          id="envelope-sustain"
          max="127"
          min="0"
          step="0.05"
          type="range"
          value="127"
        />
        <output
          aria-label="Envelope sustain"
          for="envelope-sustain"
        >
          1
        </output>
      </div>
      <div>
        <label
          for="envelope-release"
        >
          Release
        </label>
        <input
          id="envelope-release"
          max="127"
          min="0"
          step="0.05"
          type="range"
          value="25.397967959359185"
        />
        <output
          aria-label="Envelope release"
          for="envelope-release"
        >
          0.9999999999999999
        </output>
      </div>
    </section>
    <div>
      <button
        aria-label="key"
        type="button"
        value="0"
      >
        C
      </button>
      <button
        aria-label="key"
        type="button"
        value="1"
      >
        Db
      </button>
      <button
        aria-label="key"
        type="button"
        value="2"
      >
        D
      </button>
      <button
        aria-label="key"
        type="button"
        value="3"
      >
        Eb
      </button>
      <button
        aria-label="key"
        type="button"
        value="4"
      >
        E
      </button>
      <button
        aria-label="key"
        type="button"
        value="5"
      >
        F
      </button>
      <button
        aria-label="key"
        type="button"
        value="6"
      >
        Gb
      </button>
      <button
        aria-label="key"
        type="button"
        value="7"
      >
        G
      </button>
      <button
        aria-label="key"
        type="button"
        value="8"
      >
        Ab
      </button>
      <button
        aria-label="key"
        type="button"
        value="9"
      >
        A
      </button>
      <button
        aria-label="key"
        type="button"
        value="10"
      >
        Bb
      </button>
      <button
        aria-label="key"
        type="button"
        value="11"
      >
        B
      </button>
    </div>
  </form>
</div>
`;
